src/workshed/index.ts

```diff
@@ -0,0 +1,164 @@
+import { getWorkshed, haveEffect, Item, print, totalTurnsPlayed, use, visitUrl } from "kolmafia";
+import { $effect, $item, $items, AsdonMartin, clamp, DNALab, get, have, TrainSet } from "libram";
+import { dietCompleted } from "../diet";
+import { globalOptions } from "../config";
+import { potionSetupCompleted } from "../potions";
+import { estimatedTurns } from "../turns";
+import { getBestCycle, grabMedicine, offsetDefaultPieces } from "./utils";
+type WorkshedOptions = {
+  workshed: Item;
+  done?: () => boolean;
+  action?: () => void;
+};
+class GarboWorkshed {
+  private static _nextWorkshed: GarboWorkshed | null = null;
+  private static _currentWorkshed: GarboWorkshed | null = null;
+
+  workshed: Item;
+  done?: () => boolean;
+  action?: () => void;
+  constructor(options: WorkshedOptions) {
+    this.workshed = options.workshed;
+    if (options.done) this.done = options.done;
+    if (options.action) this.action = options.action;
+  }
+
+  canRemove(): boolean {
+    return this.done?.() ?? true;
+  }
+
+  use(): void {
+    if (!this.done?.()) this.action?.();
+  }
+
+  static get(item: Item | null): GarboWorkshed | null {
+    return worksheds.find(({ workshed }) => workshed === item) ?? null;
+  }
+
+  static get current(): GarboWorkshed | null {
+    GarboWorkshed._currentWorkshed ??= GarboWorkshed.get(getWorkshed());
+    return GarboWorkshed._currentWorkshed;
+  }
+
+  static get next(): GarboWorkshed | null {
+    if (get("_workshedItemUsed")) return null;
+    GarboWorkshed._nextWorkshed ??= GarboWorkshed.get(globalOptions.workshed);
+    return GarboWorkshed._nextWorkshed;
+  }
+
+  static useNext(): GarboWorkshed | null {
+    if (get("_workshedItemUsed")) return null;
+    const next = GarboWorkshed.next;
+    if (next && have(next.workshed)) {
+      use(next.workshed);
+      if (GarboWorkshed.get(getWorkshed()) === next) {
+        GarboWorkshed._nextWorkshed = null;
+        GarboWorkshed._currentWorkshed = next;
+      }
+    }
+    return GarboWorkshed._currentWorkshed;
+  }
+}
+const estimatedTurnsTomorrow = 400 + clamp((get("valueOfAdventure") - 4000) / 8, 0, 600);
+
+const worksheds = [
+  new GarboWorkshed({
+    workshed: $item`model train set`,
+    done: () => {
+      // We should always get value from the trainset, so we would never switch from it
+      return false;
+    },
+    action: () => {
+      if (!TrainSet.canConfigure()) return;
+      if (!get("trainsetConfiguration")) {
+        // Visit the workshed to make sure it's actually empty, instead of us having not yet seen it this run
+        visitUrl("campground.php?action=workshed");
+        visitUrl("main.php");
+      }
+      if (!get("trainsetConfiguration")) {
+        print("Reconfiguring trainset, as our next station is empty", "blue");
+        return TrainSet.setConfiguration(getBestCycle());
+      } else {
+        const bestTwoStations = getBestCycle().splice(0, 2);
```

Comment:
I saw this earlier and was thinking - if we do in fact find something that beats the meat station, is it optimal to do a 42-cycle instead of a 43-cycle? I'm sure we can maths this out, but we can also leave that for a future PR